packages/table/src/table2.tsx

```diff
@@ -669,11 +669,15 @@ export class Table2 extends AbstractComponent2<Table2Props, TableState, TableSna
             return false;
         }
 
-        const rowIndices = this.grid.getRowIndicesInRect(viewportRect, enableGhostCells!);
+        const rowIndices = this.grid.getRowIndicesInRect(
+            viewportRect,
+            enableGhostCells!,
+            this.columnHeaderHeight || Grid.MIN_COLUMN_HEADER_HEIGHT,
+        );
 
         const isViewportUnscrolledVertically = viewportRect != null && viewportRect.top === 0;
         const areRowHeadersLoading = hasLoadingOption(this.props.loadingOptions, TableLoadingOption.ROW_HEADERS);
-        const areGhostRowsVisible = enableGhostCells! && this.grid.isGhostIndex(rowIndices.rowIndexEnd - 1, 0);
+        const areGhostRowsVisible = enableGhostCells! && this.grid.isGhostIndex(rowIndices.rowIndexEnd, 0);
```

Comment:
I came to the same conclusion a little while ago when I was playing around with this code – that the `- 1` is not strictly necessary – so this change lgtm 👍🏽